src/components/ContactMe/ContactInfoCard/index.jsx

```diff
@@ -1,20 +1,43 @@
-import React from "react";
+import React, { useState } from "react";
 import "./styles.css";
 import Text from "../../Text";
 import Icon from "../../Icon";
 
-const ContactInfoCard = ({ text, name }) => {
+const ContactInfoCard = ({ text, name, link, linkLabel, showCopy }) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const handleCopy = () => {
+    navigator.clipboard
+      .writeText("edgar.itmember@gmail.com")
+      .then(() => {
+        setIsCopied(true);
+        setTimeout(() => setIsCopied(false), 2000); // Hide message after 2 seconds
```

Comment:
when you set timeout, you need to clear the ID, otherwise it will lead to memory leak.
Search for how to clean timeout after using, and do it inside a useEffect return.